src/evict/evict_lru.c

```diff
@@ -473,6 +474,21 @@ __evict_update_work(WT_SESSION_IMPL *session)
 		return (false);
 
 	/*
+	 * Update the eviction progress tracking.
+	 * TODO: Don't use statistics for this tracking, but it's nice to have
+	 * the statistics performant counters.
+	 */
+	evict_count = WT_STAT_READ(conn->stats, cache_eviction_clean) +
+	    WT_STAT_READ(conn->stats, cache_eviction_dirty);
+	if (evict_count > cache->saved_evict_count)
+		evict_count = evict_count - cache->saved_evict_count;
+	else
+		evict_count = 0;
```

Comment:
Please add a comment about what the progress is computing.  Between the use of previous values, resetting evict_count and hard-coded 90%, it is hard to know exactly what sort of number you're looking for.